libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/RouteExpirationHandler.kt

```diff
@@ -0,0 +1,28 @@
+package com.mapbox.navigation.base.internal.route
+
+import androidx.annotation.MainThread
+import androidx.annotation.VisibleForTesting
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.utils.internal.Time
+
+@MainThread
+object RouteExpirationHandler {
+
+    private val routesExpirationTimes = mutableMapOf<String, Long>()
```

Comment:
Why did you decide to keep refresh ttl in a separate singleton? Why not as a part of `NavigationRoute`? Is there any case when keeping data in a separate singleton works better?

Reply:
Yeah, that was my first thought as well.
But it's inconvenient in case of refresh. When you receive "refresh_ttl" (either in the original route response or refresh), it means "time in seconds since the response time". So if you receive refresh response with refresh_ttl=10, it means that the route will expire in 10 seconds after refresh, not after the original response. But if you don't receive refresh_ttl in the refresh response, you should use the old value. What's more, you can receive refresh_ttl=0 for error refresh response and in this  case you should immediately notify the user of route being invalidated.
These cases (especially the one with the error response) are not easy to handle in a model class (which is `NavigationRoute`). Imagine recreating the route object in case of an errored refresh attempt. It doesn't make sense. 
While with the singleton it's quite straightforward: we make a request, receive a response, save the expiration data. New request - update the expiration data if present. We stick to requests/responses (how it's supposed to be), not NavigationRoute objects.

Reply:
> Imagine recreating the route object in case of an errored refresh attempt. It doesn't make sense.

yep, that doesn't make much sense. What if we make those internal ttl filed mutable in `NavigaitonRoute`? 🤔

Reply:
1. It's a model class and all its fields are immutable.
2. We now have multiple ways of "modifying" a route: a) refreshRoute; b) change mutable fields.
3. I don't see why it's better than the current approach.